Thanks for the comments @imagejan! 

About testing `guessParser` for `Long` and `Float`: I'm not sure how to do this right and I'm also not sure if the `guessParser` method works as expected there. 

This test fails, because the String can and therefore will be interpreted as Integer:
```
assertEquals(3L, DefaultTableIOPlugin.guessParser("3L").apply("3L"));
```
Using a number too big for an Integer should work, I thought, but this also fails and I don't understand why:
```
assertEquals(36564573745634564L, DefaultTableIOPlugin.guessParser("36564573745634564L").apply("36564573745634564L"));
```
Here is the result:
```
java.lang.AssertionError: 
Expected :36564573745634564
Actual   :36564573745634564L
```

Similar problem with `Float` vs `Double`. Maybe I'm missing something very basic about how numbers are represented.

--- a/src/test/java/org/scijava/table/DefaultTableIOPluginTest.java
+++ b/src/test/java/org/scijava/table/DefaultTableIOPluginTest.java
@@ -255,6 +255,7 @@ public class DefaultTableIOPluginTest {
 		assertEquals(123.0, DefaultTableIOPlugin.guessParser("123.0").apply("123.0"));
 		assertEquals(-123.0, DefaultTableIOPlugin.guessParser("-123.0").apply("-123.0"));
 		assertEquals(3, DefaultTableIOPlugin.guessParser("3").apply("3"));
+		assertEquals(36564573745634564L, DefaultTableIOPlugin.guessParser("36564573745634564").apply("36564573745634564"));
 		assertEquals(1234567890.0987654321, DefaultTableIOPlugin.guessParser("1.2345678900987654E9").apply("1.2345678900987654E9"));
 		assertEquals(Double.NaN, DefaultTableIOPlugin.guessParser("NaN").apply("NaN"));
 		assertEquals(Double.NEGATIVE_INFINITY, DefaultTableIOPlugin.guessParser("-Infinity").apply("-Infinity"));

